src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -132,21 +155,40 @@ public class ProvenBlockHeaderStore : IProvenBlockHeaderStore
             this.asyncLoopFactory = asyncLoopFactory;
 
             this.performanceCounter = new BackendPerformanceCounter(dateTimeProvider);
-            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark, 100);
+            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark);
+            nodeStats.RegisterStats(this.AddComponentStats, StatsType.Component);
         }
 
-        /// <inheritdoc />
+        /// <summary>
+        /// Initializes the <see cref="ProvenBlockHeaderStore"/>.
+        /// <para>
+        /// If the <see cref="storeTip"/> is <c>null</c> the store is out of sync. This can happen when:</para>
+        /// <list>
+        ///     <item>The node crashed.</item>
+        ///     <item>The node was not closed down properly.</item>
+        /// </list>
+        /// <para>
+        /// To recover it will walk back the <see cref= "ConcurrentChain"/> until a common <see cref= "HashHeightPair"/> is found.
+        /// Then the <see cref="ProvenBlockHeaderStore"/>'s <see cref="storeTip"/> will be set to that.
+        /// </para>
+        /// </summary>
         public async Task InitializeAsync()
```

Comment:
Don't use `this.chain` even delete this class if not used other places, instead pass `ChainedHeader` in to the initialize method.

Reply:
Ok, done.